tests/integration/test_headers_and_subtotals.py

```diff
@@ -3810,3 +3811,540 @@ def it_provide_residual_test_stats_including_hs(self):
+        )
+
+    def test_cat_subdiff_x_cat_subdiff_row_bases(self, row_col_subdiff_insertions):
+        slice_ = Cube(
+            CR.CAT_SUBDIFF_X_CAT_SUBDIFF, transforms=row_col_subdiff_insertions
+        ).partitions[0]
+        row_bases = slice_.row_weighted_bases
+        # --- Bases are ALWAYS additive (even with differences) ---
+        # --- Subdiff in row 0 equals row=1 + (row=4 + row=5) ---
+        np.testing.assert_almost_equal(
+            row_bases[
+                0,
+            ],
+            row_bases[
+                1,
+            ]
+            + row_bases[
+                4,
+            ]
+            + row_bases[
+                5,
+            ],
+        )
+
+    def test_cat_subdiff_x_cat_subdiff_table_prop(self, row_col_subdiff_insertions):
+        slice_ = Cube(
+            CR.CAT_SUBDIFF_X_CAT_SUBDIFF, transforms=row_col_subdiff_insertions
+        ).partitions[0]
+
+        np.testing.assert_almost_equal(
+            slice_.table_proportions, slice_.counts / slice_.table_base
+        )
+
+    def test_cat_subdiff_x_cat_subdiff_col_prop(self, row_col_subdiff_insertions):
+        slice_ = Cube(
+            CR.CAT_SUBDIFF_X_CAT_SUBDIFF, transforms=row_col_subdiff_insertions
+        ).partitions[0]
+
+        np.testing.assert_almost_equal(
+            slice_.column_proportions, slice_.counts / slice_.column_weighted_bases
+        )
+
+    def test_cat_subdiff_x_cat_subdiff_row_prop(self, row_col_subdiff_insertions):
+        slice_ = Cube(
+            CR.CAT_SUBDIFF_X_CAT_SUBDIFF, transforms=row_col_subdiff_insertions
+        ).partitions[0]
+        np.testing.assert_almost_equal(
+            slice_.row_proportions, slice_.counts / slice_.row_weighted_bases
+        )
+
+    def test_cat_subdiff_x_cat_subdiff_pval(self, row_col_subdiff_insertions):
+        slice_ = Cube(
+            CR.CAT_SUBDIFF_X_CAT_SUBDIFF, transforms=row_col_subdiff_insertions
+        ).partitions[0]
+        np.testing.assert_equal(
+            slice_.pvals[
+                0,
+            ],
+            np.full(6, np.nan),
+        )
+        np.testing.assert_equal(slice_.pvals[:, 0], np.full(6, np.nan))
+
+    def test_cat_subdiff_x_cat_subdiff_zscore(self, row_col_subdiff_insertions):
+        slice_ = Cube(
+            CR.CAT_SUBDIFF_X_CAT_SUBDIFF, transforms=row_col_subdiff_insertions
+        ).partitions[0]
+        np.testing.assert_equal(
+            slice_.zscores[
+                0,
+            ],
+            np.full(6, np.nan),
+        )
+        np.testing.assert_equal(slice_.zscores[:, 0], np.full(6, np.nan))
+
+    def test_cat_subtot_subdiff_x_cat_subtot_subdiff_counts(
+        self, row_col_subtot_subdiff_insertions
+    ):
+        slice_ = Cube(
+            CR.CAT_SUBDIFF_X_CAT_SUBDIFF, transforms=row_col_subtot_subdiff_insertions
+        ).partitions[0]
+        # --- Subdiff in row 0 equals row=1 - (row=4 + row=5) ---
+        np.testing.assert_array_equal(
```

Comment:
I'm fighting with `black` formatting here. There's gotta be a better way, right? But black kept changing to this.

Reply:
nevermind, I think I can change from `count[1, ]` to `count[1, :]` and it is better